spec/spec_helper_local.rb

```diff
@@ -48,10 +48,24 @@ def param(type, title, param)
   it { is_expected.not_to eq(nil) }
 
   it {
-    is_expected.to run.with_params('foo', 'bar').and_return('md596948aad3fcae80c08a35c9b5958cd89')
+    is_expected.to run.with_params('foo', 'bar').and_return(
+      'md596948aad3fcae80c08a35c9b5958cd89'
+    )
   }
   it {
-    is_expected.to run.with_params('foo', 1234).and_return('md539a0e1b308278a8de5e007cd1f795920')
+    is_expected.to run.with_params('foo', 1234).and_return(
+      'md539a0e1b308278a8de5e007cd1f795920'
+    )
+  }
+  it {
+    is_expected.to run.with_params('foo', 'bar', nil, 'scram-sha-256').and_return(
```

Comment:
The `nil` parameter gives an error for me:
```
expected postgresql_password("foo", "bar", nil, "scram-sha-256", "salt") to have returned "SCRAM-SHA-256$4096:c2FsdA==$zOt2zFfUQMbpQf3/vRnYB33QDK/L7APOBHniLy39j/4=:DcW5Jp8Do7wYhVp1f9aT0cyhUfzIAozGcvzXZj+M3YI=" instead of raising ArgumentError('postgresql::postgresql_password' parameter 'sensitive' expects a Boolean value, got Undef)
```

Reply:
Hello,

Effectively `postgresql_password` have a mandatory boolean on 3 positions, I don't understand why it's work on my ci (I'm a beginner with puppet functions), 

That should be like you say here https://github.com/puppetlabs/puppetlabs-postgresql/pull/1313#discussion_r798456019

```
irb(main):001:0> require 'openssl'
=> true
irb(main):002:0> require 'base64'
=> true
irb(main):003:0>   def pg_sha256(password, salt)
    digest = digest_key(password, salt)
    'SCRAM-SHA-256$%s:%s$%s:%s' % [
      '4096',
      Base64.strict_encode64(salt),
      Base64.strict_encode64(client_key(digest)),
      Base64.strict_encode64(server_key(digest))
    ]
  end

  def digest_key(password, salt)
    OpenSSL::KDF.pbkdf2_hmac(
      password,
      salt: salt,
      iterations: 4096,
      length: 32,
      hash: OpenSSL::Digest::SHA256.new
    )
  end

  def client_key(digest_key)
    hmac = OpenSSL::HMAC.new(digest_key, OpenSSL::Digest::SHA256.new)
    hmac << 'Client Key'
    hmac.digest
    OpenSSL::Digest.new('SHA256').digest hmac.digest
  end

  def server_key(digest_key)
    hmac = OpenSSL::HMAC.new(digest_key, OpenSSL::Digest::SHA256.new)
    hmac << 'Server Key'
    hmac.digest
  end
=> :pg_sha256
=> :digest_key
=> :client_key
=> :server_key
irb(main):036:0> pg_sha256('bar', 'foo')
=> "SCRAM-SHA-256$4096:Zm9v$ea66ynZ8cS9Ty4ZkEYicwC72StsKLSwjcXIXKMgepTk=:dJYmOU6BMCaWkQOB3lrXH9OAF3lW2n3NJ26NO7Srq7U="
```

Reply:
Do you want I try to fix this on a new PR ?

Reply:
Please do. I just merged a PR which already fixed some Rubocop failures so be sure to base it on the latest commit.